### ISSUE TITLE
Question: can a revision work within a modelform.save in the admin?

### ISSUE THREAD

**Issue by BrendaH** [etianen/django-reversion#751]
I have a model called Function. This has a ForeignKey to auth.Group (the default Django Group).
In the admin on Function I display a multiselectfield to manage the users in the connected Group. These are also saved on the Group.

I know, via another issue on this project, that it is impossible to save the reverse relation between auth.Group and auth.User. So I have given up the hope of having a revision where you can revert to the Function with the Group and it's Users in that state.
However, I was hoping to be able to give a status view via the revision comments. Like, just telling which users were added or removed.

But, **I can not get reversion to display my comment**. It seems as if this whole block is just ignored or if the comment is overwritten somewhere.

I am not using a custom view. I am using a ModelForm and a ModelAdmin and was happy letting them do the work. But I am now thinking that I should be building my own view to have the actual function.save() inside the revision-block. Please tell me I'm wrong??

Things already tried to no avail:
Setting reversion.comment AFTER calling the parent form.save()
Calling self.instance.save() (so saving the function) inside this form save() and inside the reversion-block.

FunctionAdmin and GroupAdmin both extend VersionAdmin.
Function has also been registered thus: `@reversion.register(follow=(["group"]))`

admin.py:
```
class FunctionAdmin(VersionAdmin):
    actions = ['show_mailinglist']
    fieldsets = (
        (_('General'), {'fields': ('functionname', 'users', 'on_whoiswho')}),
    )
    form = FunctionAdminForm

... some irrelevant code....

    def get_form(self, request, obj=None, **kwargs):
        # only needed to get the request to the form
        form = super(FunctionAdmin, self).get_form(request, obj=obj, **kwargs)
        form.request = request
        return form
```

adminforms.py:
```
class FunctionAdminForm(forms.ModelForm):
    # Add users field that will be filled from the linked group
    users = UserModelMultipleChoiceField(
        User.objects.filter(is_active=True, peactor__isnull=False).order_by('first_name', 'last_name'),
        widget=admin.widgets.FilteredSelectMultiple(_('users'), False),
        required=False,
        help_text=_('Shows all active users'),
        label=_('users').capitalize(),
    )

    class Meta:
        model = Function
        exclude = ()

    def save(self, *args, **kwargs):
        old_users = list(self.instance.group.user_set.all())
        new_users = list(self.cleaned_data['users'])

        with reversion.create_revision():
            reversion.set_user(self.request.user)

            reversion_comment = (_("Function edited in admin.\nOld ids were %(oldvalue)s\n"
                                 "New ids are %(newvalue)s") %
                                 {'oldvalue': ','.join(str(ou) for ou in old_users),
                                 'newvalue': ','.join(str(nu) for nu in new_users)})

            # On a new Function, also create a Group with the same name
            if not hasattr(self.instance, 'group'):
                newgroup = Group.objects.create(name=self.cleaned_data['functionname'])
                self.instance.group = newgroup

            # Update the name of the Group if the Function's name was changed
            if self.instance.group.name is not self.cleaned_data['functionname']:
                    self.instance.group.name = self.cleaned_data['functionname']
                    self.instance.group.save()
                    reversion_comment = reversion_comment + _("\nName changed to " + self.cleaned_data['functionname'])

            kwargs['commit'] = True

            reversion.set_comment(reversion_comment)
            return super(FunctionAdminForm, self).save(*args, **kwargs)

    def save_m2m(self):
        self.instance.group.user_set.clear()
        self.instance.group.user_set.add(*self.cleaned_data['users'])
```

**Comment by etianen** [comment]
To override the comment in the admin integration, you need to override
log_addition and log_change in your admin class.

You can see how reversion does it here:

https://github.com/etianen/django-reversion/blob/master/reversion/admin.py

**Comment by BrendaH** [comment]
Thanks for the fast reply. I'll look into it on Tuesday. :-)

**Comment by BrendaH** [comment]
Your suggestion didn't really work for me but it pointed me in the right direction. :-)
The log_change method did not have a way to get to the message that I was putting together (or the needed info to build the message) in my `form.save()`

I ended up overwriting `ModelAdmin::construct_change_message` and in there just put my form.reversion_comment (if it was set to a non-empty string) as a return value and otherwise returned the value that `django.contrib.admin.utils.onstruct_change_message()` put together.

Closing this issue.
